recipes/wirelessStatus.lua

```diff
@@ -0,0 +1,190 @@
+- Set which application to run on widget press (add to `rc.lua`):
+
+  ```lua
+  -- wirelessStatus widget pressed function - open terminal and start `nmtui`
+  beautiful.wirelessStatus.pressed = function(self, button)
+      if button == 1 then  -- left mouse click
+          awful.spawn(terminal.." -e nmtui")
+      end
+  end
+  ```
+
+--]]
+
+
+local wibox    = require("wibox")
+local naughty  = require("naughty")
+local awful    = require("awful")
+local gears    = require("gears")
+
+local function factory(args)
+    local args     = args or {}
+
+    local wirelessStatus = {
+        widget    = args.widget or wibox.widget.imagebox(),
+        settings  = args.settings or function(self) end,
+        timeout   = args.timeout or 10,
+        pressed   = args.pressed or function(self, button) end,
+        followtag = args.followtag or false,
+        notification_preset = args.notification_preset or {},
+        showpopup = args.showpopup or "on",  -- Show notification popup while hovering above widget
+        timer     = gears.timer,
+        interface = "",
+        perc      = 0,
+        status    = "",
+        notification_text = "No connection",
+    }
+
+    -- Reset to default values
+    function wirelessStatus:reset()
+        wirelessStatus.interface = ""
+        wirelessStatus.perc = 0
+        wirelessStatus.status = ""
+        wirelessStatus.notification_text = "No connection"
+    end
+
+    -- Show notification popup
+    function wirelessStatus:show(seconds)
+        self:hide()
+
+        -- Update every time
+        self:update()
+
+        if self.followtag then
+            self.notification_preset.screen = focused()
+        end
+
+        self.notification = naughty.notify {
+            preset  = self.notification_preset,
+            text    = self.notification_text,
+            timeout = type(seconds) == "number" and seconds or self.notification_preset.timeout
+        }
+    end
+
+    -- Hide notification popup
+    function wirelessStatus:hide()
+        if self.notification then
+            naughty.destroy(self.notification)
+            self.notification = nil
+        end
+    end
+
+    function wirelessStatus:attach()
+        self.widget:connect_signal("mouse::enter", function()
+            self:show(0)
+        end)
+        self.widget:connect_signal("mouse::leave", function()
+            self:hide()
+        end)
+    end
+
+    -- Get name of wireless interface
+    function wirelessStatus:getInterface()
+        awful.spawn.easy_async_with_shell(
+            "awk 'NR==3 {printf(\"%s\\n\", $1)}' /proc/net/wireless",
+            function(stdout, exitcode)
+                -- Store interface name
+                -- Remove last character/s from string ("wlp4s0:"" -> "wlp4s0")
+                self.interface = stdout:sub(0, -3)
+
+                -- This is needed the first time there is an update
+                -- (or every time the `.interface` is equal to "")
+                self:update()
```

Comment:
Uhm. When no interface is present, this will use 100% CPU:

The update will see that the interface is the empty string and call `getInterface()`. Since no interface is present, the code above... I am not quite sure, it might error out in `stdout:sub()` when no interface is present. However, if that works out to be the empty string, this line will call `self:update()` again and we get an endless "loop" of update calls.

Reply:
I tried with some other command which has no output and `stdout:sub()` didn't throw an error. So I am adding this...
```lua
if self.interface ~= "" then
    self:update()
end
```